src/v/redpanda/admin/server.h

```diff
@@ -613,12 +613,22 @@ class admin_server {
     struct level_reset {
         using time_point = ss::timer<>::clock::time_point;
 
-        level_reset(ss::log_level level, time_point expires)
+        level_reset(ss::log_level level, std::optional<time_point> expires)
           : level(level)
           , expires(expires) {}
 
+        friend bool operator<(const level_reset& l, const level_reset& r) {
+            if (!r.expires.has_value()) {
+                return true;
+            } else if (!l.expires.has_value()) {
+                return false;
```

Comment:
question: if they both don't have an expiration value, should it return `false`?  A little confused on the logic :)

Reply:
should be totally arbitrary (as long as `val < nullopt` always evaluates to true) since `nullopt` is a proxy for "does not expire". am I missing something obvious?

Reply:
I think my question is should `!l.expires.has_value() && !r.expires.has_value()` return `false`?  Because in the logic above, `!r.expires.has_value()` always returns `true` regardless of the value of `l.expires.has_value()`

Reply:
I keep pondering on this.  On one hand, you're right it is arbitrary.  On the other nullopt < nullopt is not necessarily true.

Reply:
That's a great point. Logically arbitrary but less so semantically. I'll flip em